src/main/kotlin/uk/gov/dluhc/printapi/service/SftpService.kt

```diff
@@ -1,19 +1,54 @@
 package uk.gov.dluhc.printapi.service
 
+import com.jcraft.jsch.ChannelSftp
+import mu.KotlinLogging
+import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.integration.file.FileHeaders
 import org.springframework.integration.sftp.session.SftpRemoteFileTemplate
 import org.springframework.integration.support.MessageBuilder
 import org.springframework.stereotype.Service
 import java.io.InputStream
 
+private val logger = KotlinLogging.logger {}
+
 @Service
-class SftpService(private val sftpTemplate: SftpRemoteFileTemplate) {
+class SftpService(
+    @Qualifier("sftpInboundTemplate") private val sftpInboundTemplate: SftpRemoteFileTemplate,
+    @Qualifier("sftpOutboundTemplate") private val sftpOutboundTemplate: SftpRemoteFileTemplate,
+) {
+    companion object {
+        private const val PROCESSING_SUFFIX = ".processing"
+        private val STATUS_RESPONSE_FILE_REGEX = Regex("status-\\d{17}\\.json")
+    }
 
     fun sendFile(inputStream: InputStream, filename: String): String =
-        sftpTemplate.send(
+        sftpInboundTemplate.send(
             MessageBuilder
                 .withPayload(inputStream)
                 .setHeader(FileHeaders.FILENAME, filename)
                 .build()
         )
+
+    /**
+     * Returns list entry for all files present in the directory
+     * @param filesDirectoryPath the location of the status files
+     */
+    fun identifyFilesToBeProcessed(filesDirectoryPath: String): List<ChannelSftp.LsEntry> =
+        sftpOutboundTemplate
```

Comment:
Probably this can be done on remote SFTP server using [Spring Integration](https://stackoverflow.com/questions/59374086/spring-integration-java-dsl-remote-filter-is-not-applied-on-handle-method) - but filtering client side I expect is fine 👍

Reply:
As discussed with @Neil-Massey this is just an observation, and no action is needed to address it

Reply:
I'd like us to revisit this when we get a chance - perhaps someone can raise a story for it please?
I'd much rather the equivalent of `ls status-*.json` on the FTP server; rather then getting all the filenames and filtering them in code.
It's not a priority (we have much bigger fish to fry!), but I would like us to consider it at some point